tests/e2e/bcd_consumer_integration_test.go

```diff
@@ -548,14 +550,86 @@ func (s *BCDConsumerIntegrationTestSuite) Test09ConsumerFPFinalitySignature() {
 	s.True(finalizedBlock.Finalized)
 }
 
-// Test10BabylonFPCascadedSlashing
+func (s *BCDConsumerIntegrationTestSuite) Test10BsnFpRewards() {
+	// Ensure BSN rewards are being gathered.
+	// Initially sent to the finality contract, then sent to Babylon on an epoch boundary
+	s.Eventually(func() bool {
+		balance, err := s.cosmwasmController.QueryFinalityContractBalances()
+		if err != nil {
+			s.T().Logf("failed to query balance: %s", err.Error())
+			return false
+		}
+		if len(balance) == 0 {
+			return false
+		}
+		positiveBalance := false
+		for _, b := range balance {
+			denom := b.Denom
+			s.T().Logf("Balance of denom '%s': %s", denom, b.Amount.String())
+			// Check that the balance of the denom is greater than 0
+			if b.Amount.IsPositive() {
+				positiveBalance = true
+			} else {
+				s.T().Logf("balance of denom '%s' is not positive: %s", denom, b.Amount.String())
+			}
+		}
+		return positiveBalance
+	}, 30*time.Second, time.Second*5)
+
+	// Wait for an epoch boundary on the BSN (reward_interval = 50, so BSN height has to be a multiple of 50)
+	s.T().Logf("Waiting for an epoch boundary on the BSN")
+	rewardInterval := uint64(50)
+	s.Eventually(func() bool {
+		bsnHeight, err := s.cosmwasmController.QueryLatestBlockHeight()
+		if err != nil {
+			s.T().Logf("failed to query latest block height: %s", err.Error())
+			return false
+		}
+		s.T().Logf("BSN height: %d", bsnHeight)
+		if (bsnHeight % rewardInterval) == 0 {
+			s.T().Logf("Epoch boundary on the BSN reached")
+			return true
+		}
+		return false
+	}, 30*time.Second, time.Second*1)
+
+	// Check they end up in the Babylon "incentive" module address on an epoch boundary
+	destModule := "incentive"
+	s.T().Logf("Checking if rewards have been sent to the Babylon '%s' module address", destModule)
+	babylonPrefix := "bbn"
+	destModuleAddr, err := bech32.ConvertAndEncode(babylonPrefix, authtypes.NewModuleAddress(destModule))
+	s.NoError(err)
+	s.T().Logf("Destination module address: %s", destModuleAddr)
+
+	s.Eventually(func() bool {
+		balance, err := s.babylonController.QueryBalances(destModuleAddr)
+		if err != nil {
+			s.T().Logf("failed to query balance: %s", err.Error())
+			return false
+		}
+		if len(balance) == 0 {
+			s.T().Logf("no balance found")
+			return false
+		}
+		ibcDenom := getFirstIBCDenom(balance)
+		if ibcDenom == "" {
+			s.T().Logf("failed to get IBC denom")
+			return false
+		}
+		s.T().Logf("Balance of IBC denom '%s': %s", ibcDenom, balance.AmountOf(ibcDenom).String())
+		// Check that the balance of the IBC denom is greater than 0
+		return balance.AmountOf(ibcDenom).IsPositive()
```

Comment:
Maybe check positive is not enough, we'd better to assert the balance is increased after an epoch

Reply:
50 blocks is too long, we could use two blocks or even one (sometimes using one might cause some test-specific edge causes :P) in the test env. Yeah, a separate PR is desirable as I guess this change might break some other things.